### PR TITLE
feat: Update input 'required' indicator

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-2243]

### Description

Updates input 'required' indicator from `*` to `Required` and adds `Optional` indicator (needed for the address form).

| Address Form | Artwork Form | 
| --- | --- |
| Simulator Screen Shot - iPad mini (6th generation) - 2021-12-27 at 13 08 08|Simulator Screen Shot - iPad mini (6th generation) - 2021-12-27 at 13 08 21|

| Input | Select Form | 
| --- | --- |
| Simulator Screen Shot - iPad mini (6th generation) - 2021-12-27 at 13 15 42 | Simulator Screen Shot - iPad mini (6th generation) - 2021-12-27 at 13 14 41 |

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Updated 'required' indicator for inputs and selects from `*` to `Required` - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-2243]: https://artsyproduct.atlassian.net/browse/CX-2243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8412131615128f7383b814db9d705cfec5a73a81
</p>

**Comment by rajsam003** [comment]
Are these changes required only for the Address form?

File: src/palette/elements/Input/InputTitle.tsx [hunk 1]
```diff
@@ -13,8 +17,15 @@ export const InputTitle: React.FC<{ required?: boolean }> = ({ children: title,
       <Text variant="md" style={{ fontSize: 13, marginBottom: 2, textTransform: "uppercase" }}>
         {title}
         {!!required && (
-          <Text variant="md" color={color("blue100")}>
-            *
+          <Text variant="md" style={{ fontSize: 13, textTransform: "none" }} color={color("black60")}>
+            {" "}
```
Comment (rajsam003) [inline src/palette/elements/Input/InputTitle.tsx:21]: Have you tried using `ml=1` or `ml=0.5` instead {" "}

Comment (olerichter00) [reply src/palette/elements/Input/InputTitle.tsx:21]: Yep and it doesn't work here.

**Comment by olerichter00** [comment]
> Are these changes required only for the Address form?

Since these changes are applied to a Palette element they will be global. The specific thing about the address form is that we decided to just indicate the only optional field (as you can see in the screenshot).

**Review by rajsam003 — APPROVED** [review]
LGTM 🤝

**Pull request merged**